### PR TITLE
chore: Set up Apollo client and GQL codegen

### PR DESCRIPTION
**Pull request by connieelee**
https://wandb.atlassian.net/browse/WB-15602

- Sets up Apollo client for weave UI to make direct GQL queries to Gorilla. This enables us to send GQL mutations for exporting panels to reports.
- Sets up GQL codegen for Typescript types. If you add a new query, you must run `yarn generate` just like in core app.
- Adds a simple "get report" query in `ChildPanelReportExport`. At least one GQL document is required for `yarn generate` to work, which is in turn required for CI checks. This also allows me to demonstrate that the client works:

https://github.com/wandb/weave/assets/17016170/e96183f2-4cc8-48aa-be38-b58bf71f4fba

I also tested an `upsertView` mutation locally - it works! You'll see it in action in later PRs.

### CODE REVIEW

File: weave-js/src/entrypoint.tsx [hunk 1]
```diff
@@ -76,30 +79,32 @@ const Main = ({browserType}: MainProps) => (
...
-      </Route>
-      <Route path={`/${URL_BROWSE}/${URL_LOCAL}/:assetType?/:preview?`}>
-        <Main browserType={URL_LOCAL} />
-      </Route>
-      <Route path="/">
-        <Main />
-      </Route>
-    </Switch>
-  </Router>,
+  <ApolloProvider client={apolloClient}>
```
Comment (connieelee) [inline weave-js/src/entrypoint.tsx:82]: Recommend reviewing with whitespace hidden. This whole block just wraps what was there with an ApolloProvider

File: weave-js/wb_schema.gql [hunk 2]
```diff
@@ -0,0 +1,605 @@
+# This file is a hand-selected subset of the W&B GQL spec that is used by weave-js.
+# These type definitions are used to generate TypeScript types for GQL operations.
+# If there's a new operation you want to support in weave-js, you must add the
+# corresponding type definitions here.
```
Comment (connieelee) [inline weave-js/wb_schema.gql:4]: As mentioned [above](https://github.com/wandb/weave/pull/674#discussion_r1342969555), this file just copies an existing file, except I updated this comment text for the weave-js context

**Review by davidwallacejackson — APPROVED** [review]
Thank you for jumping through all the hoops I set up!

File: weave-js/codegen.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {CodegenConfig} from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  overwrite: true,
+  schema: './wb_schema.gql',
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/codegen.ts:5]: This is totally fine for now! Thank you Connie (: We can add some automation around this in the future. Since weave is open source, we copy a simplified schema here with just the min required edges

File: weave-js/codegen.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  documents: ['src/**/*.{ts,tsx}'],
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/codegen.ts:6]: what are the conditions for which autogen will look at a file / part of a file?

File: weave-js/wb_schema.gql (hunk 2 above)
Comment (tssweeney) [reply weave-js/wb_schema.gql:4]: hm... seeing this now - i wonder if we can just reuse the same file as we have for testing? .. cc @dannygoldstein

File: weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx [hunk 5]
```diff
@@ -36,7 +43,13 @@ export const ChildPanelExportReport = ({
     null
   );
 
+  const selectedReportQuery = useQuery(GET_REPORT, {
```
Comment (tssweeney) [inline weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx:46]: I know you are using this right now for example purposes and the main thing is to get mutations working. We should probably avoid GQL queries in weave components as much as possible. Once you are done with the mutation side, it would be nice to see if this can be articulated as a weave query.

File: weave-js/wb_schema.gql (hunk 2 above)
Comment (connieelee) [reply weave-js/wb_schema.gql:4]: If I try to point the codegen config to a file outside of the weave-js dir, this error occurs while building the integration test image: https://github.com/wandb/weave/actions/runs/6384193261/job/17326341162#step:6:286

I am in favor of using a shared schema file across weave, but need a little help resolving the build issue

File: weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx (hunk 5 above)
Comment (connieelee) [reply weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx:46]: Ah, I did work on a version of this request using weave queries at one point. I can refactor if you feel strongly about it, but truthfully I just find the direct GQL queries so much more intuitive. I know you said there are perf benefits to weave queries, but it's really hard to resist the GQL option when you consider dev experience/velocity.

I was under the impression that weave queries are only required in the context of panels (to execute input nodes), and that it would be the dev's discretion whether to use gql queries or weave queries outside of that -- as long as the two approaches are not mixed?

To be honest, I think most folks coming over from core will opt for the gql approach if it's available. The barrier is just so much lower than learning how to construct weave nodes. It might be difficult to enforce the preference for weave queries once we merge this PR.

File: weave-js/wb_schema.gql (hunk 2 above)
Comment (connieelee) [reply weave-js/wb_schema.gql]: Update -- we're now using 1 schema file across FE & BE!

File: weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx (hunk 5 above)
Comment (tssweeney) [reply weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx:46]: totally fair perspective. I believe that developers should have the power to choose what is best/easiest. My only comment on why you might NOT use the GQL approach for certain things is that if you have some thing like this:

```
Component A (assume calls useNodeValue to get some id to pass to B)
Component B (use ID in GQL query to fetch some other ID)
Component C (use ID to construct weave graph, then call useNodeValue)
```

In this case, let's say we have a `Loader` for each component... we will get a waterfall effect - where B does not load till A is done, then C doesn't load till B is done, then C needs to load the data - this is 3 serialized requests (aka waterfall problem).

Alternatively, if this was refactored such that:

```
Component A (passes weave node to B)
Component B (adds ops to incoming node, passing result to C)
Component C (adds ops to incoming node, then calls useNodeValue)
```

In this case, there is no loading state or network calls until C loads.

---

In conclusion here, i think it is good to add this client, and agree that it is easier in some cases. However, I think it is important to call out that this does allow users to have a "performance foot gun" if they design parts of their app like the above.

**Review by tssweeney — APPROVED** [review]
